### PR TITLE
Clarify conda installation with MPI

### PR DESCRIPTION
**Pull request by wmayner**
Describe how the build string in `nest-simulator=<version>=<build_string>` can be found.

### CODE REVIEW

File: doc/installation/index.rst [hunk 1]
```diff
@@ -152,7 +152,14 @@ compile NEST from source, see section** :ref:`advanced_install`.
 
              conda create --name ENVNAME -c conda-forge nest-simulator=*=mpi_openmpi*
 
-          The syntax for this install follows the pattern: ``nest-simulator=<version>=<build_string>``
+          The syntax for this install follows the pattern:
+          ``nest-simulator=<version>=<build_string>``. Build strings can be
+          found by browsing the `conda forge file list
```
Comment (terhorstd) [inline doc/installation/index.rst:157]:
```suggestion
          found by listing the available versions with
          ```
          conda search -c conda-forge nest-simulator
          ```
          or by browsing the `conda forge file list
```

File: doc/installation/index.rst [hunk 2]
```diff
@@ -152,7 +152,14 @@ compile NEST from source, see section** :ref:`advanced_install`.
...
+          <https://anaconda.org/conda-forge/nest-simulator/files>`_ (note
+          there are multiple pages). For example, to install
+          `linux-64/nest-simulator-2.20.0-mpi_openmpi_py38hb43900d_0.tar.bz2``
+          on Linux with Python 3.8, you would use the version specifier
+          ``nest-simulator=2.20.0=mpi_openmpi_py38hb43900d_0``.
```
Comment (terhorstd) [inline doc/installation/index.rst:162]: advising the user to install even a specific build, may lead to many dependency conflicts. Better let conda figure it out.
```suggestion
          one of the 2.20.x versions with MPI support by OpenMPI you would use the version specifier
          ``nest-simulator=2.20.*=*openmpi*``. The Python version dependency is automatically resolved
          if you add ``python=3`` or ``python=3.8`` depending on how specific you are tied to a Python version.
          Leaving the Python version and build identifyer open will make conda install the latest versions
          compatible with all requested packages.
```

File: doc/installation/index.rst (hunk 1 above)
Comment (wmayner) [reply doc/installation/index.rst:157]: Somehow I wasn't even aware of that command! Yes, that's definitely the way to do it.

File: doc/installation/index.rst (hunk 2 above)
Comment (wmayner) [reply doc/installation/index.rst:162]: Agreed—I didn't know that conda accepts wildcards in that way. Nice.

**Comment by wmayner** [comment]
The macOS build on Travis appears to be failing, but I'm not sure why; the changes were only to the documentation, so I'm assuming it's unrelated.

**Comment by gtrensch** [comment]
Travis passed successfully. Single reviewer is fine for this PR.

**Pull request merged**